heareval/tasks/util/luigi.py

```diff
@@ -63,6 +77,55 @@ def stage_number(self) -> int:
             raise ValueError("Unknown requires: %s" % self.requires())
 
 
+class DownloadCorpus(WorkTask):
+    """
+    Task for downloading a dataset to a specific filename
+    """
+
+    # URL to download dataset from
+    url = luigi.Parameter()
+    outfile = luigi.Parameter()
+
+    @property
+    def name(self):
+        return type(self).__name__
+
+    def run(self):
+        download_file(self.url, os.path.join(self.workdir, self.outfile))
+        with self.output().open("w") as _:
+            pass
+
+    @property
+    def stage_number(self) -> int:
+        return 0
+
+
+class ExtractCorpus(WorkTask):
+
+    infile = luigi.Parameter()
+
+    @property
+    def name(self):
+        return type(self).__name__
```

Comment:
Same as above. I'm curious if there's a way to automagically grab all parameters and put them in the name by default, unless you really specifically want to override that.